GoogleForCreators/web-stories-wp
# Dashboard: Settings DropDown

## Context

Following up earlier PR to update dashboard settings styles (#6178 ) to add in the new dropdown menu. Did this separately because we ran into a snafu with the popup that controls the dropdown menu, it's the first time the popup is getting used in the Dashboard - needed to adjust it for scroll.  

## Summary

- Swaps out the dropdown menu styles on Settings Ad Network

![settings update](https://user-images.githubusercontent.com/10720454/108018428-f17a3500-6fd4-11eb-8e69-8a6d19a789a2.gif)



## Relevant Technical Choices

- Adjust `popup` in design system to respect vertical scroll. Formerly, this component lived in the editor where there was no vertical scroll. Now, if we're going to share this structure, which it seems the editor relies on, we need to update it to factor in page scroll. I did this by using `window.pageYOffset` instead of `bodyRect.height` when calculating the `maxOffsetY` and then only using that max in the returned clamp values if it's greater than 0. The problem was that in some cases where scroll is involved the `bodyRect.height` maxOffsetY would come back with a negative number which made the returned y value `0` which was putting the popup in the wrong spot. 
- Reorganize the ad related components in `Settings` to make them easier to follow through and remove deadweight that kept the grid intact. 


## To-do
 
## User-facing changes

- Swaps out dropdown menu style on ad network form of dashboard settings 

## Testing Instructions

- See that ad network updates the same in this new style as previously. 
- Reorganized the ad management related components in `views/settings` so now in storybook you can go to `Dashboard/Views/EditorSettings/AdManagement/Default` and see the 3 components that create this section together. Just toggle the `adNetwork` knob to get different static views. 

1. If you have no adNetwork selected you will just see the initial dropdown (`AdNetwork`) and the value selected will be `None`
![Screen Shot 2021-02-16 at 11 07 11 AM](https://user-images.githubusercontent.com/10720454/108103577-5f5f4480-7047-11eb-8bb7-2d79c10510da.png)

2. If you have `AdSense` selected you will have 2 additional inputs for publisher ID and slot ID - you should be able to update these (just as before) by updating the input and hitting 'Save'. 
<img width="827" alt="Screen Shot 2021-02-16 at 11 09 17 AM" src="https://user-images.githubusercontent.com/10720454/108103686-8453b780-7047-11eb-80c3-ac01be503ac8.png">

3. If you have `Ad Manager` selected you will have 1 additional input for slot id (separate from ad sense slot id). You should still be able to update this (just as before) by typing in the input a valid string and hitting 'Save'. 
<img width="829" alt="Screen Shot 2021-02-16 at 11 10 23 AM" src="https://user-images.githubusercontent.com/10720454/108103792-ab11ee00-7047-11eb-9ab3-b8973c0c7100.png">

Because I also updated the `popup` in the design system, it is worth looking at the designsystem/components/tooltip and search to verify that I didn't regress the current use cases by updating the popup to check for window.pageOffsetY and only observing maxOffsetY if it's greater than 0.  



### QA






This PR can be tested by following these steps:

1.

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist




---


Fixes #6406
